skcosmo/preprocessing/NormalizeScaler.py

```diff
@@ -0,0 +1,70 @@
+import numpy as np
+import sklearn as sk
+from sklearn.base import TransformerMixin, BaseEstimator
+
+
+class NormalizeScaler(TransformerMixin, BaseEstimator):
+    """A more flexible scaler for features and data, that mimics
+    sklearn.preprocessing.StandardScaler but makes the mean norm of the rows
+    equal to one."""
+
+    def __init__(self, *, with_mean=True, with_std=True, per_feature=False):
+        """Initialize NormalizeScaler. Defines whether it will subtract mean
+        (with_mean=True), apply normalization (with_norm=True) and whether it will
+        normalize each feature separately (per_feature=True)."""
+
+        self.with_mean = with_mean
+        self.with_std = with_std
+        self.per_feature = per_feature
+        self.n_samples_seen_ = 0
+
+    def fit(self, X, y=None):
+        """ Compute mean and scaling to be applied for subsequent normalization. """
+
+        self.n_samples_seen_, self.n_features_ = X.shape
+        self.mean_ = np.zeros(self.n_features_)
```

Comment:
why not set to `None`? This will make a check later easier to see if it had been fit

Reply:
Not necessary. self.mean_ changes its value only if self.with_mean = True. But we may not center the components. And if then we start to check whether we have made fit using the parameter self.with_mean = True, the answer will be NO for any option, whether we have made fit or not. In my opinion, it is more logical to simply leave this parameter zero and change it later.